docs/proposals/mapper- provide-a- common- MQTT- mapper.md

```diff
@@ -0,0 +1,277 @@
+---
+title: Mapper Provide A Common MQTT Mapper
+authors: 
+
+- "@fuchendou"
+  approvers:
+
+creation-date: 2024-7-15
+last-updated: 2024-7-29
+status: implementable
+---
+
+# The proposal for Mapper Provide A Common MQTT Mapper in KubeEdge
+
+### Mapper Provide A Common MQTT Mapper
+
+        In the current hardware device market, the MQTT (Message Queuing Telemetry Transport) protocol has become a widely adopted standard. Known for its lightweight and efficient characteristics, it is particularly suitable for communication between IoT (Internet of Things) devices. However, as the types of devices and application scenarios diversify, developers face several challenges and pain points.
+
+        Firstly, different device manufacturers may implement customizations and extensions to the MQTT protocol, leading to interoperability issues between devices. Secondly, the diversity of serialization formats presents another challenge when handling device data. JSON, YAML, and XML are common serialization methods, each with its unique advantages and application scenarios.
+
+        Therefore, developing a universal MQTT can help developers simplify the device integration process, improve system flexibility and reliability.
+
+### Goals
+
+        For developers or end users of a common MQTT mapper solutions, the goals of a common MQTT mapper are:
+
+- Add a new Mapper for the MQTT protocol to the kubeedge/mappers-go project using the latest mapper-framework.
+
+- Mapper can get the message from the specified MQTT service's topic, and can parse the attribute values from the message using common serialization methods(JSON, YAML, XML). 
+
+- Provide a more scalable solution for parsing attribute values from more customized messages wherever possible.
+
+### Proposal
+
+        Providing a common MQTT mapper using the latest mapper-framework aims to help developers easily make use of mapper to solve MQTT communication, in order to simplify the device integration process, improve system flexibility and reliability.
+
+##### Scope
+
+- Development of the MQTT Mapper: Utilize the latest Mapper-Framework to add an MQTT protocol Mapper to the kubeedge/mappers-go project. Ensure the Mapper can subscribe to specified MQTT topics, receive messages, and parse them.
+
+- Message Parsing and Serialization Support: functionality to retrieve messages from MQTT service topics. Support parsing messages using JSON, YAML and XML formats to extract attribute values.
+
+##### User
+
+- Developers: Developers are needed to implement and test the MQTT Mapper.
+
+- IoT Device Manufacturers and Operators: IoT device manufacturers and operators can use this universal MQTT Mapper to manage device communications.
+
+### Design Details
+
+##### Architecture and Modules
+
+        The architecture for Mapper Provide a Common Mapper will be modular to ensure scalability and maintainability. The main components include:
```

Comment:
```suggestion
        The architecture of providing a common MQTT mapper will be modular to ensure scalability and maintainability. The main components include:
```

Reply:
Thank you for pointing out the unclear design ideas and related errors in the proposal, I will promptly revise my proposal in response to your comments.